### PR TITLE
Update nccmp package.py

### PR DESCRIPTION
**Pull request by ulmononian**
The develop version of nccmp's `package.py` is not functional, as the current nccmp recipe therein is incomplete. This PR updates nccmp `package.py` to re-establish functionality. It has been installed and tested successfully at NOAA-EMC and JCSDA.

The primary changes are (i) additional dependencies and (ii) inclusion of a cmake_args function used to properly set flags and configure cmake for the build.

### CODE REVIEW

File: var/spack/repos/builtin/packages/nccmp/package.py [hunk 1]
```diff
@@ -2,7 +2,12 @@
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
```
Comment (haampie) [inline var/spack/repos/builtin/packages/nccmp/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/nccmp/package.py [hunk 2]
```diff
@@ -15,4 +20,33 @@ class Nccmp(CMakePackage):
...
+            cflags.append("-std=c99")
+
+        if cflags:
+            args.append(self.define("CMAKE_C_FLAGS", " ".join(cflags)))
+
+        nc = self.spec["netcdf-c"]
+        if "~shared" in nc:
+            nc_pc_cmd = ["nc-config","--static","--libs"]
+            nc_flags = \
+              subprocess.check_output(nc_pc_cmd, encoding="utf8").strip()
```
Comment (haampie) [inline var/spack/repos/builtin/packages/nccmp/package.py]: Use `Executable("nc-config")(args..., output=str)` (automatically imported)

**Review by haampie — DISMISSED** [review]

File: var/spack/repos/builtin/packages/nccmp/package.py (hunk 2 above)
Comment (ulmononian) [reply var/spack/repos/builtin/packages/nccmp/package.py]: @haampie i replaced the use of the ```subprocess``` module w/ spack's ```Executable``` submodule functionality and nccmp built successfully w/ these changes. thanks for taking a look and for the useful suggestion!

File: var/spack/repos/builtin/packages/nccmp/package.py (hunk 1 above)
Comment (ulmononian) [reply var/spack/repos/builtin/packages/nccmp/package.py]: fixed duplicated header lines.

**Comment by ulmononian** [comment]
> Hi @ulmononian! I noticed that the following package(s) don't yet have maintainers:
> 
>     * nccmp
> 
> 
> Are you interested in adopting any of these package(s)? If so, simply add the following to the package class:
> 
> ```python
>     maintainers("ulmononian")
> ```
> 
> If not, could you contact the developers of this package and see if they are interested? You can quickly see who has worked on a package with `spack blame`:
> 
> ```shell
> $ spack blame nccmp
> ```
> 
> Thank you for your help! Please don't add maintainers without their consent.
> 
> _You don't have to be a Spack expert or package developer in order to be a "maintainer," it just gives us a list of users willing to review PRs or debug issues relating to this package. A package can have multiple maintainers; just add a list of GitHub handles of anyone who wants to volunteer._

added myself (@ulmononian) and @climbfuji as maintainers.

**Comment by ulmononian** [comment]
@haampie when you have a moment, would you be able to give this another look? i believe i have correctly addressed the style check test failures and other comments. thank you so much!

**Comment by aerorahul** [comment]
Dear @haampie,
Could we get a 👍🏾 👎🏾 on this PR?
Thank you

**Comment by aerorahul** [comment]
Hi @alalazo Gentle poke on this PR.

**Review by alalazo — APPROVED** [review]

**Comment by climbfuji** [comment]
Thanks @alalazo !

**Comment by ulmononian** [comment]
thank you very much, @alalazo.

**Comment by aerorahul** [comment]
Thank you much kindly @alalazo

**Pull request merged**